packages/blade/src/components/FileUpload/FileUpload.web.tsx

```diff
@@ -163,6 +163,9 @@ const _FileUpload: React.ForwardRefRenderFunction<BladeElementRef, FileUploadPro
       handleFilesChange(inputFiles);
       onChange?.({ name, fileList: allFiles });
     }
+
+    // Reset the input value to allow re-selecting the same file
+    event.target.value = '';
```

Comment:
Should this be part of the delete function of file?

Reply:
No, we don't have access to `event` there.